### PR TITLE
feat(glue): database description property

### LINKED ISSUE #27740: @aws-cdk/aws-glue-alpha: Add description to Glue table object
### Describe the feature

Glue tables can already be created with CDK:

https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-glue-alpha.Database.html

For the purpose of documentation and data governance, the database description field from CFN should be added to this:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-database-databaseinput.html#cfn-glue-database-databaseinput-description

### Use Case

This will help people that created their data lakes fully on AWS with CDK to embed documentation on IaC. Having to use a different API or manually changing database descriptions is not a feasible option for large scale users with hundreds of databases.

### Proposed Solution

Add the description field to the props of the database object.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

latest

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by msambol**
Closes #27740.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks!
The implementation looks correct, just some adjustments are needed in my opinion.
Also, can you please add validation for the property according to [this](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-database-databaseinput.html#cfn-glue-database-databaseinput-description)?

**Comment by msambol** [comment]
@lpizzinidev Updated per your recommendations. Thanks!

**Review by kaizencc — APPROVED** [review]

**Comment by kaizencc** [comment]
@msambol I am good with this PR but I want to throw this by @TheRealAmazonKendra because she is working on Glue changes that may or may not impact this PR.

File: packages/@aws-cdk/aws-glue-alpha/test/database.test.ts [hunk 1]
```diff
@@ -88,6 +111,14 @@ test('locationUri length must be <= 1024', () => {
   ).toThrow();
 });
 
+test('description length must be <= 2048', () => {
+  expect(() =>
+    new glue.Database(stack, 'Database', {
+      description: 'a'.repeat(2049),
+    }),
+  ).toThrow();
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-glue-alpha/test/database.test.ts:119]:
```suggestion
  ).toThrow(\description length must be less than or equal to 2048, got 2049\);
```
nit: add error message to test.

File: packages/@aws-cdk/aws-glue-alpha/lib/database.ts [hunk 2]
```diff
@@ -136,3 +148,9 @@ function validateLocationUri(locationUri: string): void {
     throw new Error(`locationUri length must be (inclusively) between 1 and 1024, but was ${locationUri.length}`);
   }
 }
+
+function validateDescription(description: string): void {
+  if (description.length > 2048) {
+    throw new Error(`description length must be less than or equal to 2048, but was ${description.length}`);
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-glue-alpha/lib/database.ts:154]:
```suggestion
    throw new Error(`description length must be less than or equal to 2048, got ${description.length}`);
```
extremely small nit: 'got' is a bit more standardized with the design guidelines.

**Review by scanlonp — CHANGES_REQUESTED** [review]
Should `description` be a public attribute of the `Database` class? It is added in the interface but not in the class. Also two very small nits.

Otherwise this looks great! Thank you.

**Comment by msambol** [comment]
> Should `description` be a public attribute of the `Database` class? It is added in the interface but not in the class. Also two very small nits.

@lpizzinidev mentioned this may not need to be exposed for now. I can add it if necessary.

**Comment by msambol** [comment]
@TheRealAmazonKendra Can I nudge you for a review here?

**Comment by msambol** [comment]
@kaizencc Could we push this one through?

**Review by kaizencc — DISMISSED** [review]
Done. Sorry for the delay. Thanks for the review @lpizzinidev. I thought we had some blocker on the glue construct but it's been too long now, and this is an easy win.

**Comment by paulhcsun** [comment]
Added a `pr/do-not-merge` tag on this for the time being as there is a group working on a Glue refactor (approved RFC) so we are not currently accepting any Glue PRs. We will push this through as soon as that is completed, we appreciate
 your patience @msambol

**Comment by natalie-white-aws** [comment]
Per Kendra's request, I reviewed these changes and can confirm that these changes do not conflict with the glue-alpha L2 work in flight. Safe to merge pending all other CDK service team checks without impacting our work.

**Comment by msambol** [comment]
@kaizencc can we push this through? I think an issue with the queue?

**Review by kaizencc — APPROVED** [review]

**Pull request merged**